src/source/ObjcGenerator.scala

```diff
@@ -417,6 +420,44 @@ class ObjcGenerator(spec: Spec) extends Generator(spec) {
       }
       w.wl
 
+      w.wl("- (NSDictionary *)toDict")
+      w.braced {
+        w.wl("#define _djinni_hide_null_(_o_) ((_o_)?(_o_):([NSNull null]))")
+        w.wl("")
+        w.w(s"return ").nestedN(2) {
+          w.w("@{")
+          w.w("@\"__class_name__\": [self.class description]")
+
+          for (f <- r.fields) {
+            w.w(", @\"")
+            w.w(idObjc.field(f.ident))
+            w.w("\": ")
+
+            w.w("_djinni_hide_null_(")
```

Comment:
Is nn relevant here?  The nn feature is only used for interfaces, and this feature seems to be for records, which aren't (currently) able to contain interfaces.  All datatypes in ObjC records will be non-null unless they were declared optional in the .djinni file.